test/dotcom/alerts/disruptions/subway_test.exs

```diff
@@ -81,18 +83,191 @@ defmodule Dotcom.Alerts.Disruptions.SubwayTest do
+        )
+
+      multi_active_period_alert =
+        [{beginning_of_week, active_period_1_end}, {active_period_2_start, end_of_week}]
+        |> disruption_alert()
+
+      expect(Alerts.Repo.Mock, :by_route_ids, fn _route_ids, _now ->
+        [multi_active_period_alert]
+      end)
+
+      # Exercise
+      disruptions = future_disruptions()
+
+      # Verify
+      assert Enum.count(disruptions.next_week) == 1
+      [combined_active_period_alert] = disruptions.next_week
+      assert combined_active_period_alert.active_period == [{beginning_of_week, end_of_week}]
+    end
+
+    test "combines consecutive active periods if the boundary is on consecutive days" do
+      # Setup
+      {beginning_of_week, end_of_week} = service_range_next_week()
+
+      active_period_1_end =
+        Generators.DateTime.random_time_range_date_time(
+          {beginning_of_week, DateTime.shift(end_of_week, day: -2)}
+        )
+
+      active_period_2_start =
+        Generators.DateTime.random_time_range_date_time(
+          {active_period_1_end |> ServiceDateTime.beginning_of_next_service_day(),
+           active_period_1_end
+           |> ServiceDateTime.beginning_of_next_service_day()
+           |> ServiceDateTime.end_of_service_day()}
+        )
+
+      multi_active_period_alert =
+        [{beginning_of_week, active_period_1_end}, {active_period_2_start, end_of_week}]
+        |> disruption_alert()
+
+      expect(Alerts.Repo.Mock, :by_route_ids, fn _route_ids, _now ->
+        [multi_active_period_alert]
+      end)
+
+      # Exercise
+      disruptions = future_disruptions()
+
+      # Verify
+      assert Enum.count(disruptions.next_week) == 1
+      [combined_active_period_alert] = disruptions.next_week
+      assert combined_active_period_alert.active_period == [{beginning_of_week, end_of_week}]
+    end
+
+    test "combines consecutive active periods in the middle of the active period list" do
+      # Setup
+      {beginning_of_week, end_of_week} = service_range_next_week()
+
+      active_period_0_start =
+        Generators.DateTime.random_time_range_date_time(
+          {beginning_of_week, DateTime.shift(beginning_of_week, day: 1)}
+        )
+
+      active_period_0_end =
+        Generators.DateTime.random_time_range_date_time(
+          {active_period_0_start, DateTime.shift(active_period_0_start, day: 1)}
+        )
+
+      active_period_1_start =
+        Generators.DateTime.random_time_range_date_time(
+          {DateTime.shift(active_period_0_end, day: 2), DateTime.shift(end_of_week, day: -2)}
+        )
+
+      active_period_1_end =
+        Generators.DateTime.random_time_range_date_time(
+          {active_period_1_start, DateTime.shift(end_of_week, day: -2)}
+        )
+
+      active_period_2_start =
+        Generators.DateTime.random_time_range_date_time(
+          {active_period_1_end, ServiceDateTime.end_of_service_day(active_period_1_end)}
+        )
```

Comment:
randomness on top of randomness!

I think these tests might all be more readable if we leaned into `shift`ing more and generating bespoke random times less. 

What if we pick _one_ datetime from the `random_time_range` as a start, shift that by a day or hour or whatever for the end datetime, shift _that_ by N days to define the start of the next period, and so on.... IMO this would let the values still be random, but the chosen datetimes would be easier to understand in relation to each other (and it'd probably be a lot fewer lines to write here).

Reply:
Haha I started out doing it the way you suggested, and then ran into a flaky test. The randomness solved it, believe it or not.

The day right before daylight savings time starts, there is a one-hour period of time where 23 hours later (aka `Timex.shift(days: 1)` pushes us not one but two service days forward.

I can see if `DateTime.shift(day: 1)` avoids that problem, but it would surprise me a bit if it did.

Reply:
Pesky times! Okay then.